client/internal/sender.go

```diff
@@ -31,6 +38,12 @@ type SenderCommon struct {
 	// Indicates that there is a pending message to send.
 	hasPendingMessage chan struct{}
 
+	// Indicates onMessage callback is running
+	onMessageRunning int32
```

Comment:
Maybe name this a bit differently to match the DisableScheduleSend/DisableScheduleSend names.
```suggestion
	isSendingDisabled int32
```